src/pages/dataTableUtilities/getIndicatorTableData.js

```diff
@@ -36,6 +37,37 @@ const COLUMNS = {
   },
 };
 
+/**
+ * Returns a filter method for finding a subset of an array of IndicatorAttribute objects by period.
+ * @param {Array.<IndicatorAttribute>} filterPeriod
+ * @return {Func}
+ */
+const filterByPeriod = filterPeriod => ({ period }) => period.id === filterPeriod.id;
+
+/**
+ * Returns a find method for querying an array of indicatorValue for the index of a given object.
+ * @param {IndicatorValue} findValue
+ * @return {Func}
+ */
+const findByValue = findValue => value => findValue.id === value.id;
+
+/**
+ * Query if an IndicatorValue array contains a given object.
+ * @param {Array.<IndicatorValue>} values
+ * @param {IndicatorValue} value
+ */
+const includesValue = (values, value) => {
+  const result = values.findIndex(findByValue(value)) > 0;
+  return result;
+};
+
+/**
+ * Returns a filter method for getting the intersection of two arrays of IndicatorValue objects.
+ * @param {Array.<IndicatorValue>} values
+ * @return {Func}
+ */
+const filterByValues = values => value => includesValue(values, value);
```

Comment:
could also use https://www.npmjs.com/package/set-manipulator (which we have installed)

Reply:
I spent ages trying to do it with js, I didn't know we had this! Anyway, something for a later fix (with the selectors etc.).